L1Trigger/L1THGCal/interface/HGCalTriggerTowerGeometryHelper.h

```diff
@@ -0,0 +1,53 @@
+#ifndef __L1Trigger_L1THGCal_HGCalTriggerTowerGeometryHelper_h__
+#define __L1Trigger_L1THGCal_HGCalTriggerTowerGeometryHelper_h__
+
+/** \class HGCalTriggerTowerGeometryHelper
+ *  Handles the mapping between TCs and TTs.
+ *  The mapping can be provided externally (via a mapping file)
+ *  or can be derived on the fly based on the TC eta-phi coordinates.
+ *  The bin boundaries need anyhow to be provided to establish the eta-phi coordinates
+ *  of the towers (assumed as the Tower Center for the moment)
+ */
+
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/L1THGCal/interface/HGCalTowerID.h"
+
+#include <vector>
+
+namespace l1t {
+  class HGCalTowerID;
+  class HGCalTowerCoord;
+}
+
+
+class HGCalTriggerTowerGeometryHelper {
+  public:
+    HGCalTriggerTowerGeometryHelper(const edm::ParameterSet& conf);
+
+    ~HGCalTriggerTowerGeometryHelper() {}
+
+    const std::vector<l1t::HGCalTowerCoord>& getTowerCoordinates() const;
+
+    unsigned short getTriggerTowerFromTriggerCell(const unsigned tcId, const float& eta, const float& phi) const;
+
+  private:
+    int binBinarySearch(const std::vector<double>& vec, int start, int end, const double& key) const;
+
+    std::vector<l1t::HGCalTowerCoord> tower_coords_;
+    std::map<unsigned, short> cells_to_trigger_towers_;
```

Comment:
Ordering is indeed not needed here. I opted for a `map` just based on memory concerns (given the size of the map). Anyhow I have no concrete measurement so I can move to `unordered_map` for now.

Reply:
Did you forget this change? I don't know if you looked at the timing for 200PU events, but since this map is accessed for each trigger cell, there should be a large difference of timing between a map and an unordered_map.